### ISSUE TITLE
database access variables didn't work

### ISSUE THREAD

**Issue by josimarviana** [joaofazolo/boca-docker#17]
Hi.
First, congratulations and thanks for the work.
I'm trying to configure the database access variables and it didn't work. Apparently, the system is trying to access BOCA_DB_NAME using BOCA_DB_SUPER_USER:

psql: error: FATAL:  database "superuser" does not exist

docker-compose.yml:

version: '3.8'

services:

    # web app
    boca-web:
        environment:
            # database configuration
            # privileged user password

            - BOCA_DB_NAME=boca
            - BOCA_DB_SUPER_USER=superuser
            - BOCA_DB_SUPER_PASSWORD=superp@ss
        ports:
            - 8002:80

    # online judge
    # boca-jail:

    # database
    boca-db:
        image: postgres:14-alpine
        environment:
            # database configuration
            # privileged user password
            - POSTGRES_DB=boca
            - POSTGRES_USER=superuser
            - POSTGRES_PASSWORD=superp@ss

**Comment by joaofazolo** [comment]
Hey,
I just tried and It looks like there is a bug when setting BOCA_DB_SUPER_USER. You can try with only BOCA_DB_NAME and BOCA_DB_SUPER_PASSWORD on boca-web container, and POSTGRES_DB and POSTGRES_PASSWORD on boca-db container. The superuser it's going to be 'postgres'. See if that solution works until the bug is fixed.

**Comment by rlaiola** [comment]
@joaofazolo I guess that this is the expected behavior. Let me try to explain why.

The problem happens here:

https://github.com/joaofazolo/boca-docker/blob/cec32a488244cdd61c54114df9f4146ad9cea3a7/docker/dev/web/init.sh#L63-L66

In the context of the `boca-web` container, `BOCA_DB_SUPER_USER` is the superuser that will create and change the schema of a database called `BOCA_DB_NAME`. The point is that this user should be able to authenticate in the database server (`boca-db`) even if `BOCA_DB_NAME` database does not exist.

From PostgreSQL [documentation](https://hub.docker.com/_/postgres), 

> `POSTGRES_DB` 
> This optional environment variable can be used to define a different name for the default database that is created when the image is first started. If it is not specified, then the value of `POSTGRES_USER` will be used.

In the code snippet shown above, `boca-web` tries to authenticate with the database superuser without specifying the database name. This means that it will consider that exists a database named `BOCA_DB_SUPER_USER`. If `POSTGRES_DB` is set, such a database (named `BOCA_DB_SUPER_USER`) will not be created in the PostgreSQL server and then the authentication will fail.

An alternative to ´fix´ that would be to set ``--dbname=template0`` or ``--dbname=template1``.  Note that, from a PostgreSQL database perspective, this is just a workaround because we cannot guarantee that one will not delete such a database in the future and, if so, the authentication will fail the same way.

In summary, my recommendation would be to add a note to the [documentation](https://github.com/joaofazolo/boca-docker/blob/master/tests/env/README.md) indicating that the `POSTGRES_DB` env variable should not be set in the contest of the `boca-db` service.

@josimarviana follow João´s instructions or just comment out this line:

```
# database
boca-db:
    image: postgres:14-alpine
    environment:
        # database configuration
        # privileged user password
        #- POSTGRES_DB=boca
        - POSTGRES_USER=superuser
        - POSTGRES_PASSWORD=superp@ss
```

Please let us know whether it works. 

Cheers!

**Comment by josimarviana** [comment]
Yes, this worked. Thank you very much @joaofazolo and @rlaiola.
